corehq/apps/geospatial/views.py

```diff
@@ -178,6 +178,24 @@ def post(self, request, *args, **kwargs):
             'id': geo_polygon.id,
         })
 
+    def delete(self, request, *args, **kwargs):
+        polygon_id = json.loads(request.body).get('polygon_id', None)
```

Comment:
`request.GET` won't have what you need?

Reply:
for method `delete`, it is available as `request.body`

Reply:
I believe its because of how the request is made. 
We are passing the ID as [data](https://github.com/dimagi/commcare-hq/pull/34717/files#diff-f7678da076b04a23a985a832225e6710338270977cf3f107bd1c46bc93eaae14R650) instead of putting it in the URL which is a more standard way of doing this?

Reply:
You can also omit setting the default value as `None` since this is already the default value that gets returned.

Reply:
I agree with @mkangia . Can we see if adding it as part of the URL will work?

Reply:
Agreed.  I ended up refactoring the view to follow the REST standards.

Reply:
So I split the view in `List` and `Detail` that follows REST principles and saw it being followed in HQ as well.
Addressed in https://github.com/dimagi/commcare-hq/pull/34717/commits/d3a503915347fc52ae37bfe8d1d4a7eb3693e77d with test cases in https://github.com/dimagi/commcare-hq/pull/34717/commits/9c5a55598f56affa51bb56c3665f59e2b28bbbcc